include/zephyr/net/wifi_mgmt.h

```diff
@@ -315,7 +318,9 @@ struct net_wifi_mgmt_offload {
 	 * result by the driver. The wifi mgmt part will take care of
 	 * raising the necessary event etc...
 	 */
-	int (*scan)(const struct device *dev, scan_result_cb_t cb);
+	int (*scan)(const struct device *dev,
+		    struct wifi_scan_params *params,
```

Comment:
This is a breaking change, so, need to:

* Update release notes
* Fix all Wi-Fi drivers